heartbeat/monitors/wrappers/wrappers.go

```diff
@@ -36,42 +34,44 @@ import (
 	"github.com/elastic/beats/v7/heartbeat/monitors/logger"
 	"github.com/elastic/beats/v7/heartbeat/monitors/stdfields"
 	"github.com/elastic/beats/v7/heartbeat/monitors/wrappers/monitorstate"
+	"github.com/elastic/beats/v7/heartbeat/monitors/wrappers/summarizer"
 	"github.com/elastic/beats/v7/heartbeat/scheduler/schedule"
 	"github.com/elastic/beats/v7/libbeat/beat"
 )
 
 // WrapCommon applies the common wrappers that all monitor jobs get.
 func WrapCommon(js []jobs.Job, stdMonFields stdfields.StdMonitorFields, stateLoader monitorstate.StateLoader) []jobs.Job {
-	// flapping is disabled by default until we sort out how it should work
 	mst := monitorstate.NewTracker(stateLoader, false)
-	if stdMonFields.Type == "browser" {
-		return WrapBrowser(js, stdMonFields, mst)
+	var wrapped []jobs.Job
+	if stdMonFields.Type != "browser" || stdMonFields.BadConfig {
+		wrapped = WrapLightweight(js, stdMonFields, mst)
 	} else {
-		return WrapLightweight(js, stdMonFields, mst)
+		wrapped = WrapBrowser(js, stdMonFields, mst)
 	}
+	// Wrap just the root jobs with the summarizer
+	// The summarizer itself wraps the continuations in a stateful way
+	for i, j := range wrapped {
+		j := j
+		wrapped[i] = func(event *beat.Event) ([]jobs.Job, error) {
+			s := summarizer.NewSummarizer(j, stdMonFields, mst)
+			return s.Wrap(j)(event)
+		}
+	}
+	return wrapped
```

Comment:
```suggestion
	return jobs.WrapAllSeparately(
		wrapped,
		func(j jobs.Job) jobs.JobWrapper {
			s := summarizer.NewSummarizer(j, stdMonFields, mst)
			return s.Wrap
		},
	)
```
nit: for consistency. Doing these from GH UI, so there might be some syntax errors, sorry about that.

Reply:
I've just pushed a new commit that actually removes `WrapAllSeparately` as well as some other junk this PR accumulated in `job.go`. It wasn't actually used anywhere, and I maintain that it's simpler having the logic inline IMHO.